validator/keymanager/v2/derived/derived.go

```diff
@@ -0,0 +1,307 @@
+package derived
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path"
+	"strconv"
+	"sync"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	validatorpb "github.com/prysmaticlabs/prysm/proto/validator/accounts/v2"
+	"github.com/prysmaticlabs/prysm/shared/bls"
+	"github.com/prysmaticlabs/prysm/shared/rand"
+	"github.com/prysmaticlabs/prysm/shared/roughtime"
+	"github.com/sirupsen/logrus"
+	util "github.com/wealdtech/go-eth2-util"
+	keystorev4 "github.com/wealdtech/go-eth2-wallet-encryptor-keystorev4"
+)
+
+var log = logrus.WithField("prefix", "derived-keymanager-v2")
+
+const (
+	// TimestampFileName stores a timestamp for account creation as a
+	// file for a direct keymanager account.
+	TimestampFileName = "created_at.txt"
+	// KeystoreFileName exposes the expected filename for the keystore file for an account.
+	KeystoreFileName = "keystore.json"
+	// EIPVersion used by this derived keymanager implementation.
+	EIPVersion = "EIP-2334"
+	// WithdrawalKeyDerivationPathTemplate defining the hierarchical path for withdrawal
+	// keys for Prysm eth2 validators. According to EIP-2334, the format is as follows:
+	// m / purpose / coin_type / account_index / withdrawal_key
+	WithdrawalKeyDerivationPathTemplate = "m/12381/3600/%d/0"
+	// ValidatingKeyDerivationPathTemplate defining the hierarchical path for validating
+	// keys for Prysm eth2 validators. According to EIP-2334, the format is as follows:
+	// m / purpose / coin_type / account_index / withdrawal_key / validating_key
+	ValidatingKeyDerivationPathTemplate = "m/12381/3600/%d/0/0"
+)
+
+// Wallet defines a struct which has capabilities and knowledge of how
+// to read and write important accounts-related files to the filesystem.
+// Useful for keymanager to have persistent capabilities for accounts on-disk.
+type Wallet interface {
```

Comment:
Isn't this defined somewhere else already?

Reply:
Made it a common iface